packages/serverpod/lib/src/database/expressions.dart

```diff
@@ -4,67 +4,266 @@ import 'package:serverpod/serverpod.dart';
 
 /// A database [Expression].
 class Expression {
-  /// The [String] representation of the [Expression]. Note that the precedence
+  /// The [dynamic] representation of the [Expression]. Note that the precedence
   /// of operators may not be what you think, so always use parentheses to make
   /// sure that that expressions are executed in the correct order.
-  final String expression;
+  final dynamic expression;
```

Comment:
Does this really have to be dynamic? If so, can we make it private?

Reply:
We want it dynamic because we want to apply the escaping on the actual value of that type.

But it is possible to make it private so that it isn't exposed. Then the to string method is the only way to retrieve the expression which is how we use it internally. Question is if users have used the value directly in any way?

Reply:
Changed it to a generic type and made the variable private, then created a getter marked deprecated for retrieving the expression as a string. This should be backwards compatible and tick all the boxes :)